### PR TITLE
Improve ghost buffer interface

### PR DESCRIPTION
**Pull request by simonbyrne**
WIP

### CODE REVIEW

**Comment by simonbyrne** [comment]
bors try

**Comment by simonbyrne** [comment]
bors try

**Comment by simonbyrne** [comment]
bors try

**Comment by simonbyrne** [comment]
bors try-

**Comment by sriharshakandala** [comment]
bors try

File: src/Spaces/dss.jl [hunk 1]
```diff
@@ -61,24 +73,34 @@ end
...
+dss_untransform(::Type{T}, targ, local_geometry, i, j) where {T} =
+    dss_untransform(T, targ, local_geometry[i, j])
+dss_untransform(::Type{T}, targ, local_geometry::Nothing, i, j) where {T} =
+    dss_untransform(T, targ, local_geometry)
+dss_untransform(::Type{T}, targ, local_geometry, i) where {T} =
+    dss_untransform(T, targ, local_geometry[i])
+dss_untransform(::Type{T}, targ, local_geometry::Nothing, i) where {T} =
+    dss_untransform(T, targ, local_geometry)
+
+@inline function dss_untransform(::Type{T}, targ, local_geometry)
```
Comment (sriharshakandala) [inline src/Spaces/dss.jl:85]: @inline function dss_untransform(::Type{T}, targ, local_geometry) where {T}

**Comment by simonbyrne** [comment]
bors try

**Comment by sriharshakandala** [comment]
bors try

**Comment by simonbyrne** [comment]
bors try

**Comment by simonbyrne** [comment]
bors try

**Comment by simonbyrne** [comment]
bors try

**Comment by sriharshakandala** [comment]
bors try

**Comment by sriharshakandala** [comment]
bors try

**Comment by simonbyrne** [comment]
bors try

**Comment by simonbyrne** [comment]
bors try

**Comment by sriharshakandala** [comment]
bors try

**Comment by sriharshakandala** [comment]
bors try

**Comment by simonbyrne** [comment]
bors try

**Comment by simonbyrne** [comment]
bors try

**Comment by sriharshakandala** [comment]
bors try

**Comment by simonbyrne** [comment]
bors try

**Comment by simonbyrne** [comment]
bors try

**Comment by simonbyrne** [comment]
bors try-

**Comment by simonbyrne** [comment]
bors try

**Comment by simonbyrne** [comment]
bors try-

**Comment by simonbyrne** [comment]
bors try

**Comment by simonbyrne** [comment]
Let me squash the commits

**Pull request merged**